### PR TITLE
Added 'online' option for event location

### LINKED ISSUE #325: As an authorized user/admin I want to have an opportunity to add an 'Online' location in the 'Country' field when I create a new event on the 'Profile' page so that I can organize an event online.
**Date Time**
12-11-2020 14:00
 
**Priority**
Medium 
 
**Pre-requisites**
1. Go to https://eventsexpress-test.azurewebsites.net/home/events?page=1&status=active
2. Authorize as the user/admin
3. Move to the 'Profile' page
4. Click on the 'Add event' tab  
 
**Description**
 
Each authorized user/admin must have an opportunity to change the location of a new Event in the 'Country' field to 'Online'.
 
**Acceptance criteria**
 
Given that I am an authorized user/admin,
When I create a new Event on the 'Profile' page and add the location of this Event in the 'Country' field
Then I have an opportunity to change the location of the Event to 'Online'. 
 
**Additional context**

Position in code: 
`<select class="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect" aria-invalid="Required" id="age-native-simple" name="countryId"><option value=""></option><option value="our_new_value">Online</option></select>`

### PR DESCRIPTION
**Pull request by NastiaGerasymchuk**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [x] @iuriikhrystiuk 

### Second Level Review

- [ ] @sand0 

## Summary of issue

As an authorized user/admin I want to have an opportunity to add an 'Online' location in the 'Country' field when I create a new event on the 'Profile' page so that I can organize an event online

## Summary of change

add validation for location, change service, add possibility to choose location

## Testing approach

 add an 'Online' location

## CHECK LIST
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EventsExpress.Db/Entities/EventLocation.cs [hunk 1]
```diff
@@ -10,5 +10,9 @@ public class EventLocation : BaseEntity
     {
         [Track]
         public Point Point { get; set; }
+
+        public Uri OnlineMeeting { get; set; }
```
Comment (iuriikhrystiuk) [inline EventsExpress.Db/Entities/EventLocation.cs:15]: Rest of the fields should be marked with `Track` attribute as well/

File: EventsExpress.Core/DTOs/EventDTO.cs [hunk 2]
```diff
@@ -33,14 +33,18 @@ public class EventDto
 
         public string PhotoUrl { get; set; }
 
-        public Guid PhotoId { get; set; }
+        public Guid? PhotoId { get; set; }
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/DTOs/EventDTO.cs]: Why did we neeed to make it nullable?

File: EventsExpress/ClientApp/src/components/event/event-form.js [hunk 3]
```diff
@@ -14,25 +14,58 @@ import {
...
+            }
+        }
+    }
 
     handleChange = () => {
         this.setState(state => ({
             checked: !state.checked,
         }));
     }
+    handleMapLocationChange = () => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/event/event-form.js]: These two callbacks are not used anywhere. Consider removing them togather with state.

**Comment by sand0** [comment]
I guess It would be good to fix the last code smell. It relates to the low maintainability of EventMappingProfile, and we've talked about this previously.

**Review by iuriikhrystiuk — APPROVED** [review]

**Pull request merged**